src/App.tsx

```diff
@@ -0,0 +1,138 @@
+import { Layout } from './layout';
+
+const Style = {
+  Wrapper: styled.div`
+    display: flex;
+    justify-content: space-between;
+
+    padding: 0 16px;
+    margin-top: 24px;
+  `,
+};
+
+interface AppState {
+  isOpen: boolean;
+  selectedItem: Restaurant | null;
+  filter: FilterState;
+}
+
+export interface FilterState {
+  category: CategoryOption;
+  sort: SortOption;
+}
+
+const isSortFilterType = (arg: any): arg is SortOption => {
+  return arg === 'distance' || arg === 'name';
+};
+
+const isCategoryFilterTye = (arg: any): arg is CategoryOption => {
+  return (
+    arg === '전체' ||
+    arg === '한식' ||
+    arg === '중식' ||
+    arg === '일식' ||
+    arg === '양식' ||
+    arg === '아시안' ||
+    arg === '기타'
+  );
+};
+
+export class App extends Component<any, AppState> {
+  restaurantList: Restaurant[] = getRestaurantList();
+
+  constructor(props: any) {
+    super(props);
+
+    this.state = {
+      isOpen: false,
+      selectedItem: null,
+      filter: {
+        category: '전체',
+        sort: 'name',
+      },
+    };
+  }
+
+  openModal() {
+    this.setState(() => ({ isOpen: true }));
+  }
+
+  closeModal() {
+    this.setState(() => ({ isOpen: false }));
+  }
+
+  handleClickRestaurantItem(e: React.MouseEvent<HTMLElement>) {
+    if (!(e.target instanceof HTMLElement)) return;
+
+    const selectedId = e.target.closest('li')?.id;
+    const selectedRestaurant = this.restaurantList.find(
+      ({ id }) => id === selectedId
+    );
+
+    if (selectedRestaurant === undefined) return;
+
+    this.setState(() => ({ selectedItem: selectedRestaurant }));
+    this.openModal();
+  }
+
+  handleSelectFilter(e: React.ChangeEvent<HTMLSelectElement>) {
+    const selectedOption = e.target.value;
+
+    if (isSortFilterType(selectedOption)) {
+      this.setState(() => ({
+        filter: { ...this.state.filter, sort: selectedOption },
+      }));
+    }
+
+    if (isCategoryFilterTye(selectedOption)) {
+      this.setState(() => ({
+        filter: {
+          ...this.state.filter,
+          category: selectedOption,
+        },
+      }));
+    }
+  }
+
+  render() {
+    return (
+      <Layout>
+        <GlobalStyle />
+        <Style.Wrapper>
+          <SelectBox
+            option={categoryOption}
+            handleOptionChange={this.handleSelectFilter.bind(this)}
+          />
+          <SelectBox
+            option={sortOption}
+            handleOptionChange={this.handleSelectFilter.bind(this)}
+          />
+        </Style.Wrapper>
+        <RestaurantList
+          list={filterAndSortArray(this.restaurantList, this.state.filter)}
+          handleClickRestaurantItem={this.handleClickRestaurantItem.bind(this)}
+        />
+        <Modal
+          isOpen={this.state.isOpen}
+          children={
+            this.state.selectedItem && (
+              <RestaurantDetail info={this.state.selectedItem} />
+            )
+          }
+          closeModal={this.closeModal.bind(this)}
+        />
```

Comment:
props로 children을 넘겨주시는 이유가 있을까요?

Reply:
공통 모달 컴포넌트를 만들고 안의 children만 바꿔끼워 모달을 재사용하기 위해 children을 넘겨주게 되었습니다!

Reply:
```suggestion
        <Modal
          isOpen={this.state.isOpen}
          closeModal={this.closeModal.bind(this)}
        >
        	{this.state.selectedItem && (
              <RestaurantDetail info={this.state.selectedItem} />
            )}
        </Modal>
            
```

과 같은 형식이 아닌 이유도 궁금합니다~!

Reply:
앗 이렇게 작성해줄 수 있는지 몰랐습니다...!ㅎㅎㅎ😅